src/pages/DiaryDetail/DiaryDetail.jsx

```diff
@@ -1,17 +1,96 @@
-import React from "react";
-//import styles from "../DiaryDetail/DiaryDetail.module.css";
-import { useParams } from "react-router-dom";
-import DiaryCard from "../DiaryCard/DiaryCard";
+import React, { useEffect, useState } from "react";
+import { useParams, useNavigate } from "react-router-dom";
+import { useDiaries } from "../../hooks/use-diaries";
+import styles from "../DiaryDetail/DiaryDetail.module.css";
+import { IoChevronBack } from "react-icons/io5";
+import { MdSaveAlt, MdEdit, MdOutlineDelete } from "react-icons/md";
+import NotFound from "../NotFound";
 
-export default function DiaryDetail({ diaries, onDelete, onEdit }) {
-  // showDetail
-  // delete, edit button
-  const { diaryID } = useParams();
-  const diary = diaries.find((id) => diary.id === diaryID);
+export default function DiaryDetail() {
+  const [diary, setDiary] = useState(null);
+  const [isEditing, setIsEditing] = useState(false);
+  const [newDiary, setNewDiary] = useState(null);
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const { diaries, deleteDiary, updateDiary } = useDiaries();
+
+  const handleBackClick = () => {
+    navigate(-1);
+  };
+
+  const handleDelete = () => {
+    deleteDiary(id).then(() => navigate(-1));
+  };
+
+  const toggleEdit = () => {
+    setIsEditing((prev) => !prev);
+  };
+
+  const handleEdit = (e) => {
+    const { name, value } = e.target;
+    setNewDiary((prev) => ({ ...prev, [name]: value }));
+  };
+
+  const handleSave = () => {
+    const updatedDiary = { ...diary, ...newDiary };
+    updateDiary(updatedDiary);
+    setIsEditing(false);
+  };
+
+  useEffect(() => {
+    const foundDiary = diaries.find((diary) => diary.id === id);
+
+    if (foundDiary) {
+      setDiary(foundDiary);
+      setNewDiary({ title: foundDiary.title, text: foundDiary.text });
+    } else {
+      setDiary(false);
```

Comment:
Using both `false` and `null` on a single state is chaotic..
Why not just throw an error? React will handle navigating to `NotFound` automatically

Reply:
resolved by[ ](https://github.com/Gracekangdh/DiaryApp/pull/1/commits/0299d983cb308487853f1843f29103962ba14f4e)

Reply:
resolved by[ ](https://github.com/Gracekangdh/DiaryApp/pull/1/commits/0299d983cb308487853f1843f29103962ba14f4e)